src/reconcile/rec_row.c

```diff
@@ -568,8 +568,27 @@ __rec_row_leaf_insert(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins)
 
     for (; ins != NULL; ins = WT_SKIP_NEXT(ins)) {
         WT_RET(__wt_rec_upd_select(session, r, ins, NULL, NULL, &upd_select));
-        if ((upd = upd_select.upd) == NULL)
+        if ((upd = upd_select.upd) == NULL) {
+            /*
+             * In cases where a page has grown so large we are trying to force evict it (there is
+             * content, but none of the content can be evicted), we set up fake split points, to
+             * allow the page to use update restore eviction and be split into multiple reasonably
+             * sized pages. Check if we are in this situation. The call to split with zero
+             * additional size is odd, but split takes into account saved updates in a special way
+             * for this case already.
+             */
+            if (!upd_select.upd_saved || !F_ISSET(r, WT_REC_EVICT) || !__wt_rec_need_split(r, 0))
```

Comment:
I don't understand the reason for testing `WT_REF_EVICT` in this clause.

Reply:
Not any more, I think. This was my first attempt to fix teh overflow key case - I saw that it was in checkpoint, and I figured that we shouldn't be making checkpoint reconciliations split and do update-restore.

Reply:
130e2af